src/WeakValueDict.jl

```diff
@@ -0,0 +1,222 @@
+# Weak value dictionaries
+#
+# This file is a modification of base/weakkeydict.jl from the Julia project.
+
+import Base: isempty, setindex!, getkey, length, iterate, empty, delete!, pop!, get, sizehint!, copy
+"""
+    WeakValueDict([itr])
+
+`WeakValueDict()` constructs a hash table where the values are weak
+references to objects which may be garbage collected even when
+used as keys in a hash table.
+"""
+mutable struct WeakValueDict{K,V} <: AbstractDict{K,V}
+    ht::Dict{K,WeakRef}
+    lock::ReentrantLock
+    finalizer::Function
+    dirty::Bool
+
+    # Constructors mirror Dict's
+    function WeakValueDict{K,V}() where V where K
+        t = new(Dict{K,Any}(), ReentrantLock(), identity, 0)
```

Comment:
I know `0 == false` in Julia, but still, perhaps like this?
```suggestion
        t = new(Dict{K,Any}(), ReentrantLock(), identity, false)
```